trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/model/OutputCapabilities.java

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Ericsson
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0 which
+ * accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+
+/**
+ * Contributes to the model used for TSP swagger-core annotations.
+ */
+public interface OutputCapabilities {
+
+    /**
+     * @return The name.
+     */
+    @JsonProperty("canCreate")
```

Comment:
Should each capabilities parameter be independently optional (required=false) or only the parent 'capabilities'?

Or none? I'm not sure what is the intended outcome, the ADR did not have any optional in the javascript example. In the Java code (IDataProviderDescriptor) the capabilities field is not nullable.

Reply:
Not sure if we should make each one optional. I guess it could save some bytes over the TSP. However, on the client side of the TSP, we would have to handle if fields are not there. Either way, let me know what you prefer. It will affect the related PRs (tsp-typesctipt-client/tsp-python-client)

BTW, the class DataProviderDescriptorSerializer determines if the capabilities are serialized or not. IDataProviderDescriptor is the Trace Compass internal implementation/API.

Reply:
Ok, I think we agree to make each subfield optional. It's required=false by default so nothing to change in this file.